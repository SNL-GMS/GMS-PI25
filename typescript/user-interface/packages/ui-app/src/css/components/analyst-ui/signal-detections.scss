.ian-signal-detections-gl-container,
.signal-detection-table-wrapper {
  height: 100%;
}

.signal-detection-panel {
  height: 100%;
  display: flex;
  flex-direction: column;
}

.edge-SD-row .table-cell {
  background-color: var(--gms-background, $gms-background);
  filter: brightness(0.5);
}

.sd-cell {
  &--open-associated .table-cell {
    background-color: var(--analyst-open-event, $analyst-open-event);
  }
  &--open-associated > .table-cell .table-cell__value {
    color: var(--open-associated-text-color, var(--gms-main-inverted, $gms-main-inverted));
  }

  &--complete-associated .table-cell {
    background-color: var(--analyst-complete, $analyst-complete);
  }
  &--complete-associated > .table-cell .table-cell__value {
    color: var(--complete-associated-text-color, var(--gms-main-inverted, $gms-main-inverted));
  }

  &--other-associated .table-cell {
    background-color: var(--analyst-other, $analyst-other);
  }
  &--other-associated > .table-cell .table-cell__value {
    color: var(--other-associated-text-color, var(--gms-main-inverted, $gms-main-inverted));
  }

  &--unassociated .table-cell {
    background-color: var(--analyst-unassociated, $analyst-unassociated);
  }
  &--unassociated > .table-cell .table-cell__value {
    color: var(--unassociated-text-color, var(--gms-main-inverted, $gms-main-inverted));
  }

  &--deleted-sd .table-cell {
    background-color: var(--deleted-sd-color, $deleted-sd-color);
  }
  &--deleted-sd > .table-cell .table-cell__value {
    color: var(--deleted-sd-text-color, var(--gms-main-inverted, $gms-main-inverted));
  }
}

@container (max-width: 1100px) {
  .signal-detection-panel {
    .counter__count--prefix {
      display: none;
    }
    .counter__entry + .counter__entry {
      margin-left: 1rem;
    }
    .counter__count {
      margin-left: 0.25rem;
    }
  }
}

@container (max-width: 800px) {
  .signal-detection-panel {
    .counter__entry + .counter__entry {
      margin-left: 0.5rem;
    }
    .counter__label {
      display: none;
    }
  }
}
