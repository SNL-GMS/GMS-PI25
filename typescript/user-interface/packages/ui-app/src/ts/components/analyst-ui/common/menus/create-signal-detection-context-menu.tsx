/* eslint-disable react/destructuring-assignment */
import { Menu } from '@blueprintjs/core';
import { MenuItem2 } from '@blueprintjs/popover2';
import type { ImperativeContextMenuOpenFunc } from '@gms/ui-core-components';
import { ImperativeContextMenu, useImperativeContextMenuCallback } from '@gms/ui-core-components';
import { useKeyboardShortcutConfigurations } from '@gms/ui-state';
import { UILogger } from '@gms/ui-util';
import React from 'react';

import {
  formatHotkeyString,
  getKeyboardShortcutCombos
} from '~common-ui/components/keyboard-shortcuts/keyboard-shortcuts-util';

const logger = UILogger.create(
  'GMS_LOG_SIGNAL_DETECTION_CONTEXT_MENU',
  process.env.GMS_LOG_WORKFLOW_COMMANDS
);

export interface CreateSignalDetectionContextMenuCallbacks {
  readonly createSignalDetectionCb: ImperativeContextMenuOpenFunc<
    CreateSignalDetectionContextMenuProps
  >;
}

export type CreateSignalDetectionContextMenuGetOpenCallbackFunc = (
  callbacks: CreateSignalDetectionContextMenuCallbacks
) => void;

export interface CreateSignalDetectionContextMenuProps {
  channelId: string;
  timeSecs: number;
  currentPhase?: string;
  defaultSignalDetectionPhase?: string;
  createSignalDetection: (
    stationId: string,
    channelName: string,
    timeSecs: number,
    phase?: string,
    isTemporary?: boolean
  ) => void;
}

/**
 * Props are passed in by the createSignalDetectionCb call when it gets triggered.
 * This is a render prop of the imperative context menu, which will pass the props set
 * by that callback.
 * Note, unlike for normal components, props may be undefined, so we have to guard against that
 *  */
export function CreateSignalDetectionMenuContent(props: CreateSignalDetectionContextMenuProps) {
  const keyboardShortcutConfig = useKeyboardShortcutConfigurations();

  return (
    <Menu className="create-sd-menu">
      <MenuItem2
        text="Create signal detection associated to a waveform with current phase"
        label={
          keyboardShortcutConfig?.clickEvents?.createSignalDetectionWithCurrentPhase
            ? formatHotkeyString(
                `${keyboardShortcutConfig.clickEvents.createSignalDetectionWithCurrentPhase?.combos[0]}+click`
              )
            : ''
        }
        onClick={() => {
          logger.info(
            'Create signal detection associated to a waveform with current phase (e + click)'
          );
          try {
            props?.createSignalDetection(
              props?.channelId,
              undefined,
              props?.timeSecs,
              props?.currentPhase
            );
          } catch (e) {
            logger.error(e);
          }
        }}
      />
      <MenuItem2
        text="Create signal detection associated to a waveform with default phase"
        label={
          keyboardShortcutConfig?.clickEvents?.createSignalDetectionWithDefaultPhase
            ? formatHotkeyString(
                `${keyboardShortcutConfig.clickEvents.createSignalDetectionWithDefaultPhase?.combos[0]}+click`
              )
            : ''
        }
        onClick={() => {
          logger.info(
            'Create signal detection associated to a waveform with default phase (alt + e + click)'
          );
          try {
            props?.createSignalDetection(
              props?.channelId,
              undefined,
              props?.timeSecs,
              props?.defaultSignalDetectionPhase
            );
          } catch (e) {
            logger.error(e);
          }
        }}
      />
      <MenuItem2
        text="Create signal detection not associated to a waveform with current phase"
        label={
          keyboardShortcutConfig?.clickEvents
            ?.createSignalDetectionNotAssociatedWithWaveformCurrentPhase
            ? formatHotkeyString(
                getKeyboardShortcutCombos(
                  keyboardShortcutConfig?.clickEvents
                    ?.createSignalDetectionNotAssociatedWithWaveformCurrentPhase,
                  keyboardShortcutConfig
                )[0]
              )
            : ''
        }
        onClick={() => {
          logger.info(
            'Create signal detection not associated to a waveform with current phase (shift + e + click)'
          );
          try {
            props?.createSignalDetection(
              props?.channelId,
              undefined,
              props?.timeSecs,
              props?.currentPhase,
              true
            );
          } catch (e) {
            logger.error(e);
          }
        }}
      />
      <MenuItem2
        text="Create signal detection not associated to a waveform with default phase"
        label={
          keyboardShortcutConfig?.clickEvents
            ?.createSignalDetectionNotAssociatedWithWaveformDefaultPhase
            ? formatHotkeyString(
                `${keyboardShortcutConfig.clickEvents.createSignalDetectionNotAssociatedWithWaveformDefaultPhase?.combos[0]}+click`
              )
            : ''
        }
        onClick={() => {
          logger.info(
            'Create signal detection not associated to a waveform with default phase (shift + alt + e + click)'
          );
          try {
            props?.createSignalDetection(
              props?.channelId,
              undefined,
              props?.timeSecs,
              props?.defaultSignalDetectionPhase,
              true
            );
          } catch (e) {
            logger.error(e);
          }
        }}
      />
    </Menu>
  );
}

export function CreateSignalDetectionMenu(props: {
  getOpenCallback: CreateSignalDetectionContextMenuGetOpenCallbackFunc;
}) {
  const { getOpenCallback } = props;

  const [createSignalDetectionCb, setCreateSignalDetectionCb] = useImperativeContextMenuCallback<
    CreateSignalDetectionContextMenuProps
  >();

  React.useEffect(() => {
    getOpenCallback({
      createSignalDetectionCb
    });
  }, [getOpenCallback, createSignalDetectionCb]);

  return (
    <ImperativeContextMenu<CreateSignalDetectionContextMenuProps>
      getOpenCallback={setCreateSignalDetectionCb}
      content={CreateSignalDetectionMenuContent} // Props are passed in by the createSignalDetectionCb call when it gets triggered
    />
  );
}
