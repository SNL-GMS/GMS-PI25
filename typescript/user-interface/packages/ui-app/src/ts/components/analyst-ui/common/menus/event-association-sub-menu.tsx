import { MenuItem2 } from '@blueprintjs/popover2';
import type { EventTypes } from '@gms/common-model';
import { findPreferredEventHypothesisByStage } from '@gms/common-model/lib/event/util';
import {
  selectActionTargetSignalDetectionIds,
  selectAreSelectedSdsAllDeleted,
  selectOpenEvent,
  selectOpenIntervalName,
  useAppSelector,
  useAssociateSignalDetections,
  useDetermineActionTargetsByType,
  useSetActionType,
  useUnassociateSignalDetections
} from '@gms/ui-state';
import React from 'react';

import { formatHotkeyString } from '~common-ui/components/keyboard-shortcuts/keyboard-shortcuts-util';

import { useGetSignalDetectionKeyboardShortcut } from '../hotkey-configs/signal-detection-hotkey-configs';

export interface EventAssociationSubmenuProps {
  readonly keyPrefix: string;
}

/**
 * Displays a blueprint context menu for event association.
 * This is a submenu for use in other menus
 *
 * @returns the event association context menu
 */

export function EventAssociationSubmenu({ keyPrefix }: EventAssociationSubmenuProps): JSX.Element {
  const hotkeyCombo = useGetSignalDetectionKeyboardShortcut();
  const setActionType = useSetActionType();
  const determineActionTargetsByType = useDetermineActionTargetsByType();

  // Only show the first hotkey in the menu
  const associateSelectedSignalDetectionsHotkey =
    hotkeyCombo.associateSelectedSignalDetections?.combos[0];

  const unassociateSelectedSignalDetectionsHotkey =
    hotkeyCombo.unassociateSelectedSignalDetections?.combos[0];

  const associateSignalDetections = useAssociateSignalDetections();
  const unassociateSignalDetections = useUnassociateSignalDetections();

  const currentOpenEvent = useAppSelector(selectOpenEvent);
  const actionTargetSignalDetectionIds = useAppSelector(selectActionTargetSignalDetectionIds);

  const openIntervalName: string = useAppSelector(selectOpenIntervalName);
  const areSelectedSdsAllDeleted = useAppSelector(selectAreSelectedSdsAllDeleted);
  const currentOpenEventPreferredEventHypothesis: EventTypes.EventHypothesis = findPreferredEventHypothesisByStage(
    currentOpenEvent,
    openIntervalName
  );

  /** Memoized handler for onMouseLeave */
  const clearActionType = React.useCallback(() => {
    setActionType(null);
  }, [setActionType]);

  /** Memoized onClick handler for associating a signalDetection */
  const associateOnClick = React.useCallback(() => {
    associateSignalDetections(actionTargetSignalDetectionIds);
  }, [actionTargetSignalDetectionIds, associateSignalDetections]);

  /** Memoized onClick handler for unassociating a signalDetection */
  const unassociateOnClick = React.useCallback(() => {
    unassociateSignalDetections(actionTargetSignalDetectionIds, false);
  }, [actionTargetSignalDetectionIds, unassociateSignalDetections]);

  /** Memoized onClick handler for rejecting a signalDetection association */
  const rejectOnClick = React.useCallback(() => {
    unassociateSignalDetections(actionTargetSignalDetectionIds, true);
  }, [actionTargetSignalDetectionIds, unassociateSignalDetections]);

  // array of SD Id's that are associated to the open event, if any
  const sdIdsAssociatedToOpenEvent: string[] = currentOpenEventPreferredEventHypothesis?.associatedSignalDetectionHypotheses
    ? currentOpenEventPreferredEventHypothesis.associatedSignalDetectionHypotheses.map(
        sd => sd.id.signalDetectionId
      )
    : [];
  // array of SD Id's that are deleted in the open event, if any
  const sdIdsDeletedFromOpenEvent: string[] =
    currentOpenEventPreferredEventHypothesis &&
    currentOpenEvent?.rejectedSignalDetectionAssociations
      ? currentOpenEvent.rejectedSignalDetectionAssociations.map(sd => sd.id)
      : [];

  // array of the selected SD Id's that AREN'T associated to open event
  const unassociatedSdIds: string[] = actionTargetSignalDetectionIds.filter(
    id => !sdIdsAssociatedToOpenEvent.includes(id)
  );
  // array of the selected SD Id's that ARE associated to open event
  const associatedSdIds: string[] = actionTargetSignalDetectionIds.filter(id =>
    sdIdsAssociatedToOpenEvent.includes(id)
  );
  // array of the selected SD Id's that are deleted from the open event
  const deletedSdIds: string[] = actionTargetSignalDetectionIds.filter(id =>
    sdIdsDeletedFromOpenEvent.includes(id)
  );

  const isRejectedOrDeleted: boolean =
    currentOpenEventPreferredEventHypothesis?.deleted ||
    currentOpenEventPreferredEventHypothesis?.rejected;

  const menuOptionAssociateToCurrentlyOpenEvent = (
    <MenuItem2
      text={`Associate ${determineActionTargetsByType('associate').length} to open event`}
      labelElement={formatHotkeyString(associateSelectedSignalDetectionsHotkey)}
      onClick={associateOnClick}
      data-cy="associate-to-open"
      disabled={
        !currentOpenEvent || // no open event
        unassociatedSdIds.length === 0 || // all selected SD's are already associated
        isRejectedOrDeleted // open event is deleted or rejected
      }
      onMouseEnter={() => setActionType('associate')}
      onMouseLeave={clearActionType}
      key={`${keyPrefix}-assocopen`}
    />
  );
  const menuOptionUnassociateFromCurrentlyOpenEvent = (
    <MenuItem2
      text={`Unassociate ${determineActionTargetsByType('unassociate').length} from open event`}
      labelElement={formatHotkeyString(unassociateSelectedSignalDetectionsHotkey)}
      onClick={unassociateOnClick}
      data-cy="unassociate-to-open"
      disabled={
        !currentOpenEvent || // no open event
        associatedSdIds.length === 0 || // no selected SD's are associated
        isRejectedOrDeleted // open event is deleted or rejected
      }
      onMouseEnter={() => setActionType('unassociate')}
      onMouseLeave={clearActionType}
      key={`${keyPrefix}-unassocopen`}
    />
  );
  const menuOptionRejectFromCurrentlyOpenEvent = (
    <MenuItem2
      text={`Reject ${determineActionTargetsByType('reject associate').length} from open event`}
      onClick={rejectOnClick}
      data-cy="reject-to-open"
      disabled={
        !currentOpenEvent || // no open event
        deletedSdIds.length === actionTargetSignalDetectionIds.length || // all selected SD's are already deleted
        associatedSdIds.length === 0 || // no selected SD's are associated
        isRejectedOrDeleted // open event is deleted or rejected
      }
      onMouseEnter={() => setActionType('reject associate')}
      onMouseLeave={clearActionType}
      key={`${keyPrefix}-rejectopen`}
    />
  );

  const isDisabled = !currentOpenEvent || isRejectedOrDeleted || areSelectedSdsAllDeleted;

  return (
    <MenuItem2
      text="Event association"
      disabled={isDisabled}
      data-cy="association-menu"
      key={`${keyPrefix}-event-association`}
    >
      {menuOptionAssociateToCurrentlyOpenEvent}
      {menuOptionUnassociateFromCurrentlyOpenEvent}
      {menuOptionRejectFromCurrentlyOpenEvent}
    </MenuItem2>
  );
}
