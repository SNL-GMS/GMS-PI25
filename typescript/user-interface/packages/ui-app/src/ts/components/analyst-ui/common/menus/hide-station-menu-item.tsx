import { MenuItem2 } from '@blueprintjs/popover2';
import React from 'react';

import type { HideStationMenuItemProps } from './types';

/**
 * Menu item intended to be used with the waveform display to show/hide stations
 *
 * @param props
 * @constructor
 */
export function HideStationMenuItem(props: HideStationMenuItemProps) {
  const { stationName, hideStationCallback, showHideText, disabled } = props;
  return (
    <MenuItem2
      disabled={disabled}
      data-cy={`hide-${stationName}`}
      className="hide-station-menu-item"
      onClick={hideStationCallback}
      text={showHideText ?? `Hide ${stationName}`}
      title={disabled ? 'Waveform display and an interval must be open to show or hide' : ''}
    />
  );
}
