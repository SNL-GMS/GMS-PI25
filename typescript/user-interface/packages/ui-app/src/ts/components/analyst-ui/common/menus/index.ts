export { HideStationMenuItem } from './hide-station-menu-item';
