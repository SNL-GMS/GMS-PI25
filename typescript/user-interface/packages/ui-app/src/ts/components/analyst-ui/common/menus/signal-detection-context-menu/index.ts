export type { UpdateSignalDetection } from './signal-detection-context-menu';
export { SignalDetectionContextMenu } from './signal-detection-context-menu';
export type { SignalDetectionContextMenuContentProps } from './signal-detection-context-menu-content';
export { SignalDetectionContextMenuContent } from './signal-detection-context-menu-content';
