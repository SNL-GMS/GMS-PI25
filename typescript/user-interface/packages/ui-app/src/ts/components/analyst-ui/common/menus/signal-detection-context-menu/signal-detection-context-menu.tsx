import type {
  ImperativeContextMenuGetOpenCallbackFunc,
  ImperativeContextMenuOpenFunc
} from '@gms/ui-core-components';
import { ImperativeContextMenu } from '@gms/ui-core-components';
import type { UpdateSignalDetectionArgs } from '@gms/ui-state';
import { useSetSignalDetectionActionTargets } from '@gms/ui-state';
import React from 'react';

import type { SignalDetectionDetailsProps } from '~analyst-ui/common/dialogs/signal-detection-details/types';

import type { SignalDetectionContextMenuContentProps } from './signal-detection-context-menu-content';
import { SignalDetectionContextMenuContent } from './signal-detection-context-menu-content';

/**
 * Defines the callback signature for updating signal detection.
 */
export type UpdateSignalDetection = (args: UpdateSignalDetectionArgs) => void;

export type SignalDetectionPhaseUpdate = (sdIds: string[], phase: string) => void;

export interface SignalDetectionContextMenuProps {
  getOpenCallback: ImperativeContextMenuGetOpenCallbackFunc<SignalDetectionContextMenuContentProps>;
  signalDetectionDetailsCb?: ImperativeContextMenuOpenFunc<SignalDetectionDetailsProps>;
  setCreateEventMenuCb?: (visibility: boolean, lat: number, lon: number) => void;
}

/**
 * Displays the Signal Detection Details Context Menu.
 *
 * @params props @see {@link ImperativeContextMenuGetOpenCallbackFunc}
 */
export const SignalDetectionContextMenu = React.memo(function SignalDetectionDetailsContextMenu(
  props: SignalDetectionContextMenuProps
): JSX.Element {
  const { getOpenCallback, signalDetectionDetailsCb } = props;
  const setSignalDetectionActionTargets = useSetSignalDetectionActionTargets();

  const content = React.useCallback(
    (p: SignalDetectionContextMenuContentProps) => (
      <SignalDetectionContextMenuContent
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...p}
        signalDetectionDetailsCb={signalDetectionDetailsCb}
      />
    ),
    [signalDetectionDetailsCb]
  );

  return (
    <ImperativeContextMenu<SignalDetectionContextMenuContentProps>
      content={content}
      getOpenCallback={getOpenCallback}
      onClose={() => setSignalDetectionActionTargets([])}
    />
  );
});
