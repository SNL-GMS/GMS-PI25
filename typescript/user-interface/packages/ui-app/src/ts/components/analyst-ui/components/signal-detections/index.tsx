import { IanDisplays, toDisplayTitle } from '@gms/common-model/lib/displays/types';
import { nonIdealStateWithSpinner } from '@gms/ui-core-components';
import React from 'react';

import type { SignalDetectionsComponentProps } from './types';

const SignalDetectionsLazy = React.lazy(async () =>
  import(/* webpackChunkName: 'ui-app-signal-detections' */ './signal-detections-component').then(
    module => ({
      default: module.SignalDetectionsComponent
    })
  )
);

/** Top-level component for the Signal Detections display, lazy-loaded */
export function SignalDetections(props: SignalDetectionsComponentProps) {
  return (
    <React.Suspense
      fallback={nonIdealStateWithSpinner(
        toDisplayTitle(IanDisplays.SIGNAL_DETECTIONS),
        'Loading Display'
      )}
    >
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <SignalDetectionsLazy {...props} />
    </React.Suspense>
  );
}
