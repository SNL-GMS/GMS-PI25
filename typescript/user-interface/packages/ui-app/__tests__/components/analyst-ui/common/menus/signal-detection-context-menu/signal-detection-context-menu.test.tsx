import type { CommonTypes } from '@gms/common-model';
import { processingAnalystConfigurationData } from '@gms/common-model/__tests__/__data__';
import { eventData } from '@gms/common-model/__tests__/__data__/event/event-data';
import { signalDetectionsData } from '@gms/common-model/__tests__/__data__/signal-detections/signal-detection-data';
import type { AppState } from '@gms/ui-state';
import { AnalystWorkspaceTypes, getStore } from '@gms/ui-state';
import { appState } from '@gms/ui-state/__tests__/test-util';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';

import type { SignalDetectionContextMenuContentProps } from '~analyst-ui/common/menus/signal-detection-context-menu';
import { SignalDetectionContextMenuContent } from '~analyst-ui/common/menus/signal-detection-context-menu';

// Mock data setup
const loading = false;
const isLoading = () => {
  return loading;
};
// eslint-disable-next-line @typescript-eslint/no-magic-numbers
const now = 1234567890 / 1000;
const timeRange: CommonTypes.TimeRange = {
  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
  startTimeSecs: now - 3600,
  endTimeSecs: now
};
const mockShortcuts = processingAnalystConfigurationData.keyboardShortcuts;
jest.mock('@gms/ui-state', () => {
  const actual = jest.requireActual('@gms/ui-state');
  return {
    ...actual,
    useAppSelector: jest.fn((stateFunc: (state: AppState) => any) => {
      const state: AppState = appState;
      state.app.workflow.openIntervalName = 'AL1';
      state.app.analyst.openEventId = 'eventId';
      return stateFunc(state);
    }),
    useGetSignalDetections: jest.fn(() => ({
      data: signalDetectionsData,
      isLoading: isLoading()
    })),
    useViewableInterval: jest.fn(() => [timeRange, jest.fn]),
    useKeyboardShortcutConfigurations: jest.fn(() => mockShortcuts)
  };
});

jest.mock('../../../../../../../common-model/src/ts/event/util', () => {
  const actual = jest.requireActual('../../../../../../../common-model/src/ts/event/util');
  return {
    ...actual,
    findPreferredEventHypothesisByStage: jest.fn(() => [eventData.eventHypotheses[0]])
  };
});

jest.mock(
  '../../../../../../src/ts/components/analyst-ui/common/hotkey-configs/signal-detection-hotkey-configs',
  () => {
    const actual = jest.requireActual(
      '../../../../../../src/ts/components/analyst-ui/common/hotkey-configs/signal-detection-hotkey-configs'
    );
    return {
      ...actual,
      useGetSignalDetectionKeyboardShortcut: jest.fn().mockReturnValue({
        associateSelectedSignalDetections: {
          description: 'Associate Selected Signal Detections To Open Event',
          helpText: 'Associate selected signal detection(s) to the currently open event',
          combos: ['ctrl+g'],
          tags: ['associate', 'signal', 'detection'],
          categories: ['Signal Detections List Display']
        },
        unassociateSelectedSignalDetections: {
          description: 'Unassociate Selected Signal Detections To Open Event',
          helpText: 'Unassociate selected signal detection(s) to the currently open event',
          combos: ['alt+g'],
          tags: ['unassociate', 'signal', 'detection'],
          categories: ['Signal Detections List Display']
        },
        currentPhaseLabel: {
          description: 'Set Selected Signal Detections Phase Label to Current Phase',
          helpText: 'Set phase label for selected signal detection(s) to the current phase.',
          combos: ['shift+e'],
          tags: ['set', 'update', 'phase label', 'current phase'],
          categories: ['Signal Detections List Display']
        }
      })
    };
  }
);

describe('Signal Detection Context Menu', () => {
  const props: SignalDetectionContextMenuContentProps = {
    keyPrefix: 'test-sd',
    signalDetectionDetailsCb: jest.fn(),
    measurementMode: {
      mode: AnalystWorkspaceTypes.WaveformDisplayMode.MEASUREMENT,
      entries: {
        [signalDetectionsData[0].id]: true
      }
    },
    setMeasurementModeEntries: jest.fn()
  };

  it('renders', () => {
    const store = getStore();
    const { setMeasurementModeEntries } = props;
    const { container } = render(
      <Provider store={store}>
        <SignalDetectionContextMenuContent
          keyPrefix="test-sd"
          signalDetectionDetailsCb={props.signalDetectionDetailsCb}
          measurementMode={props.measurementMode}
          setMeasurementModeEntries={setMeasurementModeEntries}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
