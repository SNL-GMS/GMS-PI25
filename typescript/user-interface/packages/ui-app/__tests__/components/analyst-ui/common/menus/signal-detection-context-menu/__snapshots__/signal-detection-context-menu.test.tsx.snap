// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signal Detection Context Menu renders 1`] = `
<div>
  <ul
    class="bp4-menu signal-detection-table__context-menu"
    role="menu"
  >
    <li
      class=""
      role="none"
    >
      <a
        class="bp4-menu-item bp4-disabled"
        data-cy="show-sd-details"
        label="(alt + click)"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="bp4-fill bp4-text-overflow-ellipsis"
        >
          Open signal detection details
        </div>
        <span
          class="bp4-menu-item-label"
        >
          (alt + click)
        </span>
      </a>
    </li>
    <li
      class="bp4-submenu"
      role="none"
    >
      <span
        aria-haspopup="true"
        class="bp4-popover2-target"
        role="menuitem"
        tabindex="0"
      >
        <a
          class="bp4-menu-item bp4-disabled"
          data-cy="association-menu"
          role="none"
          tabindex="-1"
        >
          <div
            class="bp4-fill bp4-text-overflow-ellipsis"
          >
            Event association
          </div>
          <span
            aria-hidden="true"
            class="bp4-icon bp4-icon-caret-right bp4-submenu-icon"
            icon="caret-right"
          >
            <svg
              data-icon="caret-right"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </a>
      </span>
    </li>
    <li
      class=""
      role="none"
    >
      <a
        class="bp4-menu-item bp4-disabled"
        data-cy="set-phase"
        label="(ctrl + shift + e)"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="bp4-fill bp4-text-overflow-ellipsis"
        >
          Set 0 phases
        </div>
        <span
          class="bp4-menu-item-label"
        >
          (ctrl + shift + e)
        </span>
      </a>
    </li>
    <li
      class=""
      role="none"
    >
      <a
        class="bp4-menu-item bp4-disabled"
        data-cy="rephaseCurrent-sd"
        label="(shift + e)"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="bp4-fill bp4-text-overflow-ellipsis"
        >
          Set 0 to current phase: null
        </div>
        <span
          class="bp4-menu-item-label"
        >
          (shift + e)
        </span>
      </a>
    </li>
    <li
      class=""
      role="none"
    >
      <a
        class="bp4-menu-item bp4-disabled"
        data-cy="rephaseDefault-sd"
        label="(alt + shift + e)"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="bp4-fill bp4-text-overflow-ellipsis"
        >
          Set 0 to default phase: P
        </div>
        <span
          class="bp4-menu-item-label"
        >
          (alt + shift + e)
        </span>
      </a>
    </li>
    <li
      class=""
      role="none"
    >
      <a
        class="bp4-menu-item bp4-disabled"
        data-cy="show-fk"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="bp4-fill bp4-text-overflow-ellipsis"
        >
          Show 0 FK
        </div>
      </a>
    </li>
    <li
      class=""
      role="none"
    >
      <a
        class="bp4-menu-item bp4-disabled"
        data-cy="delete-sd"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="bp4-fill bp4-text-overflow-ellipsis"
        >
          Delete 0
        </div>
      </a>
    </li>
    <li
      class=""
      role="none"
    >
      <a
        class="bp4-menu-item bp4-popover2-dismiss"
        data-cy="menu-item-export"
        role="menuitem"
        tabindex="0"
      >
        <span
          class="bp4-menu-item-icon"
        >
          <span
            aria-hidden="true"
            class="bp4-icon bp4-icon-export"
            icon="export"
            tabindex="-1"
          >
            <svg
              data-icon="export"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M4 6c.28 0 .53-.11.71-.29L7 3.41V11c0 .55.45 1 1 1s1-.45 1-1V3.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71l-4-4C8.53.11 8.28 0 8 0s-.53.11-.71.29l-4 4A1.003 1.003 0 004 6zm11 5c-.55 0-1 .45-1 1v2H2v-2c0-.55-.45-1-1-1s-1 .45-1 1v3c0 .55.45 1 1 1h14c.55 0 1-.45 1-1v-3c0-.55-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <div
          class="bp4-fill bp4-text-overflow-ellipsis"
        >
          Export
        </div>
      </a>
    </li>
  </ul>
</div>
`;
